src/sentry/runner/commands/configoptions.py

```diff
@@ -279,6 +279,10 @@ def sync(ctx: click.Context) -> None:
                         # we can safely continue.
                         if options.is_set_on_disk(opt.name):
                             continue
+                        # isset() will pull from local cache, which gets updated with every get() call.
+                        # configoptions is only concerned with removing the option from the db.
+                        elif not options.set_on_db(opt.name):
```

Comment:
We could also change `isset()` so that it behaves correctly. Currently the docstring for `isset` claims

>  Check if a key has been set to a value and not inheriting from its default.

Which doesn't match its current behavior. We could have isset compare the value from the store to the option default. If the store value and default are the same, the option is not 'set'.

Reply:
I originally wanted to do that but serveral other files and methods use `issset()`, worried I could break their functionality

Reply:
The other issue is that if this is what we use to compare options, `configoptions` would never actually remove options from the DB if the DB value is just the default value.

Reply:
> I originally wanted to do that but serveral other files and methods use issset(), worried I could break their functionality

We could review those callsites and see if their behaviour would change.

> The other issue is that if this is what we use to compare options, configoptions would never actually remove options from the DB if the DB value is just the default value.

If the value is set in the store and equal to the default, wouldn't we still be able to check the option's update channel?